templates/keepalived.conf.j2

```diff
@@ -58,6 +58,36 @@ vrrp_track_file {{ name }} {
 {% endfor %}
 {% endif %}
 
+{% if keepalived_track_processs is defined %}
+{% for name, details in keepalived_track_processs.items() %}
+vrrp_track_process {{ name }} {
+  process {{ details.process }} # see man 5 keepalived.conf for quotes handling
+  weight {{ details.weight | default(1) | int }} # multiply with weight in process (default: 1, i.e. literal value from process)
+  {% if details.param_match is defined and details.param_match in ['initial', 'partial'] %}
```

Comment:
`{% if details.param_match is defined` ... no param_match is not mandatory, but it accepts just these values
```
# keepalived -D --config-test -f /tmp/keepalived.conf ; echo $?
(/tmp/keepalived.conf: Line 15) Invalid param_match type partialx - ignoring
0
```
looks like the jinja2-side validation is required.
Doing this in the task before the cfg is deployed with a `ansible.builtin.assert` would be possible too
Splitting up the conditions and writing a comment on wrong settings could be a option too
```diff
diff --git a/templates/keepalived.conf.j2 b/templates/keepalived.conf.j2
index de383fb..582dcbe 100644
--- a/templates/keepalived.conf.j2
+++ b/templates/keepalived.conf.j2
@@ -63,8 +63,12 @@ vrrp_track_file {{ name }} {
 vrrp_track_process {{ name }} {
   process {{ details.process }} # see man 5 keepalived.conf for quotes handling
   weight {{ details.weight | default(1) | int }} # multiply with weight in process (default: 1, i.e. literal value from process)
-  {% if details.param_match is defined and details.param_match in ['initial', 'partial'] %}
+  {% if details.param_match is defined %}
+  {% if details.param_match in ['initial', 'partial'] %}
   param_match {{ details.param_match }}
+  {% else %}
+  # param_match value {{ details.param_match }} is not valid
+  {% endif %}
   {% endif %}
   {% if details.quorum is defined %}
   quorum {{ details.quorum | int }}
```

Maybe the last proposal is the most obvious and less intrusive ...

Reply:
```
# keepalived -D --config-test -f /tmp/keepalived.conf ; echo $?
(/tmp/keepalived.conf: Line 15) Invalid param_match type partialx - ignoring
0
```
As the config test is in fact eloquent, I will remove the second part of the condition and leave it up to the user to type the right value

Reply:
602bad0

Reply:
I trust your judgment.

If we have an issue later, we can add a comment then!

Reply:
removing the validation from the jinja template (`param_match in ['initial', 'partial']`) and leaving this to the keepalived makes sense looking at the warning; and in case the mapping is extended by keepalived with other values, the role enforces no restriction snd needs no update ... and as you mentioned: on should trust the users of the role as mature  and trained in technical terms.